### PR TITLE
improved specifications for keyword arguments of run method

### LINKED ISSUE #3190: Allow specification of ProgressBar keyword arguments in the AnalysisBase run method
The ProgressBar (subclass of tqdm) in the AnalysisBase doesn't currently take in keyword arguments nor does it give much access to the underlying functionality of tqdm. In particular, the position keyword argument cannot be changed to adjust the location of the bar on screen. When spreading analyses across multiple processes, the rendered bars are constantly overwriting each other.

https://github.com/MDAnalysis/mdanalysis/blob/6282385644d672cf7023584347e35048935ab429/package/MDAnalysis/analysis/base.py#L186

### PR DESCRIPTION
**Pull request by amdnsr**
Fixes #3190

Changes made in this Pull Request:
 - added `kwargs` in `run `method
 - made `verbose` a part of `kwargs` before passing it to `ProgressBar `constructor

PR Checklist
------------
 - [ ] Tests?
 - [x] Docs?
 - [x] CHANGELOG updated?
 - [x] Issue raised/referenced?

### CODE REVIEW

**Comment by amdnsr** [comment]
I preferred not to delete the argument `verbose`, because this method might have been already used with `verbose` as the positional argument (fourth argument) i.e. without the keyword `verbose`.

**Comment by amdnsr** [comment]
Please provide me some feedback on the pull request.
Thank you

**Comment by IAlibay** [comment]
@amdnsr one of the @MDAnalysis/gsoc-mentors will review your PR as soon as they can. Reviews often take a few days as all developers on the project are volunteers, so we often have our own work priorities that come before MDAnalysis.

File: package/MDAnalysis/analysis/base.py [hunk 1]
```diff
@@ -179,13 +179,15 @@ def run(self, start=None, stop=None, step=None, verbose=None):
...
+        kwargs['verbose'] = verbose
 
         self._setup_frames(self._trajectory, start, stop, step)
         logger.info("Starting preparation")
         self._prepare()
+
         for i, ts in enumerate(ProgressBar(
                 self._trajectory[self.start:self.stop:self.step],
-                verbose=verbose)):
+                **kwargs)):
```
Comment (lilyminium) [inline package/MDAnalysis/analysis/base.py:190]: May I ask why you didn't simply `verbose=verbose, **kwargs`? That way you don't have to modify the kwargs dictionary.

**Review by lilyminium — CHANGES_REQUESTED** [review]
Thanks for looking at this, @amdnsr. Please add a test so you can know that your changes work. You should also document `kwargs` in the docstring and update CHANGELOG (message of what your fix is, and your GitHub username on top). As this is your first contribution to MDAnalysis, please also add yourself to AUTHORS.

**Comment by IAlibay** [comment]
@amdnsr the deadline for GSoC applications is rapidly approaching. Are you still interested in completing this?

File: package/CHANGELOG [hunk 2]
```diff
@@ -13,6 +13,22 @@ The rules for this file:
   * release numbers follow "Semantic Versioning" http://semver.org
 
 ------------------------------------------------------------------------------
+04/06/21 amdnsr
```
Comment (IAlibay) [inline package/CHANGELOG:16]: There's already a 2.0.0 entry below, you should only need to add your entry there.

Comment (amdnsr) [reply package/CHANGELOG:16]: Apologies, I did not get it. Should I add my changes just above the following line?
09/05/19 IAlibay, richardjgowers

  * 0.20.1

Comment (IAlibay) [reply package/CHANGELOG:16]: No, there's already an existing 2.0.0 entry just below, this is where it should go.

File: package/MDAnalysis/analysis/base.py (hunk 1 above)
Comment (amdnsr) [reply package/MDAnalysis/analysis/base.py:193]: Hey, I tried that, and yes, I won't have to modify the `kwargs` dictionary in the run function of base.py file, but the `kwargs` dictionary in the `__init__` function of log.py file is expecting verbose to be a key in its `kwargs` dictionary, but in the suggested way, verbose will be a named parameter, and not a key in the `kwargs` dictionary of the `__init__` function.
So, I would have to instead do the modifications over there.
This modification seemed cleaner to me, thus, I did it this way, but if the suggested way (`verbose=verbose, **kwargs`) is preferred, I will do it that way instead.

Comment (amdnsr) [reply package/MDAnalysis/analysis/base.py:193]: Also, please guide me on creating the tests. From what I understand, I need to create a function at the end of the testsuite/MDAnalysisTests/analysis/test_base.py file, that calls the ProgressBar, with some kwargs, which will in turn be passed to tqdm, and consequently, affect the position of the ProgressBars.
But, I can't figure out how and what set of methods to call to do it, and how to verify whether my function is behaving as expected.
Thanks

Comment (lilyminium) [reply package/MDAnalysis/analysis/base.py:193]: Have a look at the [tqdm documentation](https://github.com/tqdm/tqdm#documentation) for possible parameters you could pass through, and the existing ProgressBar tests in MDAnalysis for suggestions on how you could test for progress bar appearance. We also have a [brief guide](https://userguide.mdanalysis.org/stable/testing.html#writing-new-tests) for writing tests in the user guide.

Comment (lilyminium) [reply package/MDAnalysis/analysis/base.py:193]: > Hey, I tried that, and yes, I won't have to modify the `kwargs` dictionary in the run function of base.py file, but the `kwargs` dictionary in the `__init__` function of log.py file is expecting verbose to be a key in its `kwargs` dictionary, but in the suggested way, verbose will be a named parameter, and not a key in the `kwargs` dictionary of the `__init__` function.
> So, I would have to instead do the modifications over there.

I don't think I understand what you mean here. Could you show me your suggested changes? If you just pass in `verbose=verbose, **kwargs`, the log.ProgressBar class will consider both of them to be in `kwargs` due to [dictionary unpacking](https://docs.python.org/3/tutorial/controlflow.html#keyword-arguments)

File: package/AUTHORS [hunk 3]
```diff
@@ -155,7 +155,7 @@ Chronological list of authors
 2021
   - Aditya Kamath
   - Leonardo Barneschi
-  
+  - Ahmad Nasir
 External code
```
Comment (IAlibay) [inline package/AUTHORS:159]: Please make sure that you keep the empty line between the author list and the "External code" heading.

**Review by IAlibay — CHANGES_REQUESTED** [review]
@amdnsr just a reminder that there's only a few days left before the GSoC deadline. You'll need to address @lilyminium's [comments](https://github.com/MDAnalysis/mdanalysis/pull/3191/files#r608204313) in order for this PR to progress.

**Comment by IAlibay** [comment]
@amdnsr are you still looking to work on this?

**Comment by IAlibay** [comment]
Closing PR as stale.

**Pull request closed without merging**